play-json/jvm/src/main/scala/play/api/libs/json/jackson/JacksonJson.scala

```diff
@@ -30,21 +29,25 @@ import scala.collection.mutable.{ ArrayBuffer, ListBuffer }
  * {{{
  *   import com.fasterxml.jackson.databind.ObjectMapper
  *
- *   val mapper = new ObjectMapper().registerModule(PlayJsonModule)
+ *   val jsonParseSettings = JsonParserSettings()
+ *   val mapper = new ObjectMapper().registerModule(PlayJsonModule(jsonParseSettings))
  *   val jsValue = mapper.readValue("""{"foo":"bar"}""", classOf[JsValue])
  * }}}
  */
-object PlayJsonModule extends SimpleModule("PlayJson", Version.unknownVersion()) {
+class PlayJsonModule(parserSettings: JsonParserSettings) extends SimpleModule("PlayJson", Version.unknownVersion()) {
   override def setupModule(context: SetupContext): Unit = {
-    context.addDeserializers(new PlayDeserializers)
+    context.addDeserializers(new PlayDeserializers(parserSettings))
     context.addSerializers(new PlaySerializers)
   }
 }
 
+object PlayJsonModule extends PlayJsonModule(JsonParserSettings.settings)
```

Comment:
Having to race against anything else on the classpath initializing this and therefore locking it to whatever it's currently set to feels wrong.

I propose the follow:

* deprecate public `PlayJsonModule` (the object)
* change it to use `JsonParserSettings()`, the (safe) default, so it's behavior is stable
* change `JacksonJson` (which is `package[json]`) from object to class so we don't introduce a race there either (you can then revert back the lazy modifiers)

Reply:
About the `JacksonJson`, what I mean was:

```scala
private lazy val jackson = JacksonJson(JsonParseSettings.settings)
```

In the end, maybe what we need is a way to initialize `mapper` and `jsonFactory` in `JacksonJson`.

Reply:
No, personally I wouldn't have any lazy vals in objects and just create `JacksonJson` every time you need to parse.

Reply:
> create JacksonJson every time you need to parse.

Which implies that we are creating `ObjectMapper` every time also and that is not a good idea (it is expensive to created). As far as I know, `ObjectMapper` is thread-safe and meant to be re-used across the lifetime of the application.

Reply:
Then I guess make `EnvReads` and `EnvWrites` classes and deprecate the objects?

Reply:
Actually, why do we need to mutate object `JsonParserSettings`?  Can't we just wire a different `JsonParserSettings`?

Reply:
@marcospereira If you want to get this in for RC1, I suggest leaving out the global mutable state part.

Reply:
> Actually, why do we need to mutate object JsonParserSettings? Can't we just wire a different JsonParserSettings instance?

That is why I was suggesting to instead have a way to initialize and mutate `ObjectMapper` in `JacksonJson`. Looks more reasonable.

> @marcospereira If you want to get this in for RC1, I suggest leaving out the global mutable state part.

We can do RC1 without this change.

Reply:
I mean a non-static way to wire it.

Basically, I would prefer not introducing more global mutable state.  So if a user wants to customise this behavior, they forgo these statically initialised object API, they create their own `JsonParserSettings` (or `JacksonJson`, `EnvReads`/`EnvWrites`) and goes about their business.  WDYT?

Reply:
Yeah, but that implies more profound changes. The main entry point for play-json API is to call `Json.toJson(something)`, which uses global state (`ObjectMapper`, `JsonFactory`, etc.). Change this is way out of the scope for this PR.

What we need here is to find a way to:

1. Fix the big decimal parsing.
2. And configure the bounds for the big decimal parsing.

I agree that having mutable global state is not ideal, but how to achieve that right now without changing too much of play-json?